### ISSUE TITLE
[rapidjson] Build error

### ISSUE THREAD

**Issue by Chienvoxx** [microsoft/vcpkg#25512]
Package: rapidjson[core]:x64-windows-static -> 2022-06-28
**Host Environment**
- Host: x64-windows
- Compiler: MSVC 19.32.31329.0
-    vcpkg-tool version: 2022-06-17-9268e366206712e38102b28dbd1617697a99ff2e
    vcpkg-scripts version: 881c1b04a 2022-06-30 (7 hours ago)

**To Reproduce**
`vcpkg install assimp`

**Failure logs**
```
-- Downloading https://github.com/Tencent/rapidjson/archive/232389d4f1012dddec4ef84861face2d2ba85709.tar.gz -> Tencent-rapidjson-232389d4f1012dddec4ef84861face2d2ba85709.tar.gz...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'manifests' = off
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Downloading https://github.com/Tencent/rapidjson/archive/232389d4f1012dddec4ef84861face2d2ba85709.tar.gz
Error: Failed to download from mirror set:
File does not have the expected hash:
             url : [ https://github.com/Tencent/rapidjson/archive/232389d4f1012dddec4ef84861face2d2ba85709.tar.gz ]
       File path : [ C:\libs\vcpkg\downloads\Tencent-rapidjson-232389d4f1012dddec4ef84861face2d2ba85709.tar.gz.32592.part ]
   Expected hash : [ 60bfbfe4884122aa9ae8755531da18ce4a793bb217c06474dd8c5afe0f2f5df280c40be2f303c1740f5af14c7ad5cc2a57f08de5205cc7ab27d6166b0f49130d ]
     Actual hash : [ 0d7d751179abdaa6ebf6167d522651a2d13bc024d20c7e3f775c7397a8aab4cd866a6c91a55521ad7847e910822fcf982625c7308c74f5df663e6fd81336c9fc ]


[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(788): 
[DEBUG] Time in subprocesses: 0 us
[DEBUG] Time in parsing JSON: 1 us
[DEBUG] Time in JSON reader: 0 us
[DEBUG] Time in filesystem: 1533 us
[DEBUG] Time in loading ports: 0 us
[DEBUG] Exiting after 1.391 s (1385019 us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:24 (message):
      
      Failed to download file with error: 1
      If you use a proxy, please check your proxy setting. Possible causes are:
      
      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.
      
      2. You are using Fiddler. Currently a bug (https://github.com/microsoft/vcpkg/issues/17752)
         will set HTTPS_PROXY to `https://fiddler_address:port` which lead to problem 1 above.
         Workaround is open Windows 10 Settings App, and search for Proxy Configuration page,
         Change `http=address:port;https=address:port` to `address`, and fill the port number.
      
      3. Your proxy's remote server is out of service.
      
      In future vcpkg releases, if you are using Windows, you no longer need to set
      HTTP(S)_PROXY environment variables. Vcpkg will simply apply Windows IE Proxy
      Settings set by your proxy software. See (https://github.com/microsoft/vcpkg-tool/pull/49)
      and (https://github.com/microsoft/vcpkg-tool/pull/77)
      
      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:249 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
  ports/rapidjson/portfile.cmake:2 (vcpkg_from_github)
  scripts/ports.cmake:146 (include)



```

**Additional context**

**Comment by Osyotr** [comment]
Duplicate of #25508, will be fixed in #25509

**Comment by JonLiu1993** [comment]
Duplicate of https://github.com/microsoft/vcpkg/issues/25508
